### PR TITLE
Create Interface to upload contacts to Active Campaign

### PR DESCRIPTION
**Pull request by fp-ps**

### CODE REVIEW

File: backend/de.metas.marketing/activecampaign/src/main/java/de/metas/marketing/gateway/activecampaign/restapi/RestService.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+		{
+			try
+			{
+				log.debug("Performing [{}] on [{}] for request [{}]", httpMethod.name(), resourceURI, request);
+				retry = false;
+				response = createRestTemplate().exchange(resourceURI, httpMethod, request, responseType);
+			}
+			catch (final RateLimitExceededException rateLimitEx)
+			{
+				Loggables.withLogger(log, Level.ERROR)
```
Comment (metas-ts) [inline backend/de.metas.marketing/activecampaign/src/main/java/de/metas/marketing/gateway/activecampaign/restapi/RestService.java:135]: is this an error? if so, how do we make sure to not exceed the rate limit under normal circumstances?

File: backend/de.metas.marketing/activecampaign/src/main/java/de/metas/marketing/gateway/activecampaign/restapi/exception/RateLimitService.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+		final int retryAfter = NumberUtils.asInteger(retryAfterValue.get(0), 0);
+
+		return Optional.of(Duration.ofSeconds(retryAfter));
+	}
+
+	public void waitForLimitReset(@NonNull final Duration retryAfterDuration)
+	{
+		Loggables.withLogger(log, Level.DEBUG).addLog("RateLimitService.waitForLimitReset() with retryAfter seconds: {}", retryAfterDuration.getSeconds());
+
+		final int maxSecondsToWait = sysConfigBL.getIntValue(MAX_SECONDS_TO_WAIT_FOR_ACTIVE_CAMPAIGN_LIMIT_RESET, 3600);
```
Comment (metas-ts) [inline backend/de.metas.marketing/activecampaign/src/main/java/de/metas/marketing/gateway/activecampaign/restapi/exception/RateLimitService.java:71]: what'S the purpose of this sysconfig?

File: backend/de.metas.marketing/activecampaign/src/main/java/de/metas/marketing/gateway/activecampaign/restapi/exception/RateLimitService.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+		if (retryAfterDuration.getSeconds() > maxSecondsToWait)
+		{
+			throw new AdempiereException("Retry After duration is too far in the future! aborting!")
```
Comment (metas-ts) [inline backend/de.metas.marketing/activecampaign/src/main/java/de/metas/marketing/gateway/activecampaign/restapi/exception/RateLimitService.java:75]: how shall the users react if this happens?

**Review by metas-ts — COMMENTED** [review]
..some preliminary questions..

File: backend/de.metas.marketing/base/src/main/java/de/metas/marketing/base/spi/PlatformClient.java [hunk 4]
```diff
@@ -40,7 +40,8 @@
 
 	List<RemoteToLocalSyncResult> syncCampaignsRemoteToLocal(List<Campaign> campaigns);
```
Comment (metas-ts) [inline backend/de.metas.marketing/base/src/main/java/de/metas/marketing/base/spi/PlatformClient.java:41]: it seems that when exchanging data with the CMS-platform, we pretty much rely on having enough memory.
- do the platforms have pagination mechanisms that we use?
- do we split the data that we want to send in chunks?

**Review by metas-ts — COMMENTED** [review]
please check my further comments; they might inform the next steps, even if we integrate this as a WI

File: backend/de.metas.marketing/activecampaign/src/main/java/de/metas/marketing/gateway/activecampaign/restapi/RestService.java (hunk 1 above)
Comment (cp-ps) [reply backend/de.metas.marketing/activecampaign/src/main/java/de/metas/marketing/gateway/activecampaign/restapi/RestService.java:135]: use `info` instead

File: backend/de.metas.marketing/activecampaign/src/main/java/de/metas/marketing/gateway/activecampaign/restapi/exception/RateLimitService.java (hunk 3 above)
Comment (fp-ps) [reply backend/de.metas.marketing/activecampaign/src/main/java/de/metas/marketing/gateway/activecampaign/restapi/exception/RateLimitService.java:75]: Will change the error message to inform the user how much time he needs to wait before syncing again

**Pull request merged**